atom/browser/atom_resource_dispatcher_host_delegate.cc

```diff
@@ -56,7 +56,8 @@ void HandleExternalProtocolInUI(
 
   GURL escaped_url(net::EscapeExternalHandlerValue(url.spec()));
   auto callback = base::Bind(&OnOpenExternal, escaped_url);
-  permission_helper->RequestOpenExternalPermission(callback, has_user_gesture);
+  permission_helper->RequestOpenExternalPermission(callback, has_user_gesture,
```

Comment:
we want to pass the render_frame_id/render_process_id into `HandleExternalProtocolInUI` to get the real requesting render frame host so we don't always get the main frame render frame host in `WebContentsPermissionHelper::RequestPermission`